src/Components/CreateImageWizard/steps/Details/index.tsx

```diff
@@ -24,13 +27,32 @@ const DetailsStep = () => {
   const dispatch = useAppDispatch();
   const blueprintName = useAppSelector(selectBlueprintName);
   const blueprintDescription = useAppSelector(selectBlueprintDescription);
+  const distribution = useAppSelector(selectDistribution);
+  const arch = useAppSelector(selectArchitecture);
+  const targetEnvironments = useAppSelector(selectImageTypes);
+
+  useEffect(() => {
+    if (!blueprintName) {
+      dispatch(changeBlueprintName(generateDefaultName()));
+    }
+  }, [dispatch, blueprintName]);
   const handleNameChange = (
     _event: React.FormEvent<HTMLInputElement>,
     name: string
   ) => {
     dispatch(changeBlueprintName(name));
   };
 
+  const generateDefaultName = () => {
+    const date = new Date();
+
+    const day = date.getDate().toString().padStart(2, '0');
+    const month = date.toLocaleString('en-US', { month: 'long' });
+    const year = date.getFullYear();
+    const environments = targetEnvironments.join(' ');
+    return `${distribution} ${arch} ${environments} ${day} ${month} ${year}`;
```

Comment:
This name does not pass the validation, does it?

Reply:
should it? if we know for sure that the default name is valid, why should check the validation?

Reply:
I was just thinking we disabled spaces, but we didn't, so this passes the validation.
My concern was this would be judged as invalid name and wouldn't allow user to proceed, but I was wrong ;)
Disregard! :)